### ISSUE TITLE
String to ObjectId automatic conversion - Mongodb

### ISSUE THREAD

**Issue by FranckBontemps** [mikro-orm/mikro-orm#401]
**Describe the bug**
I use mikro-orm as part of an event storage system, sometimes an event sends a Payload containing an ObjectId like a string from another system.
Mikro-orm automatically turns it into an ObjectId. This is a behavior that forces us to adapt our serialization.

**To Reproduce**
I use:
```
@Entity()
export class DataContainer extends BaseEntity {

  constructor(data: any) {
    super();
    this.data = data;
  }

  @Property()
  data: any; // typing used to have some edge cases
}
```

The test:
```
test('do not automatically convert string to ObjectId in the all cases', async () => {
    const dataContainer = new DataContainer({
      partnerIdFromAnotherSystem: '0000007b5c9c61c332380f78',
    });

    expect(dataContainer.data.partnerIdFromAnotherSystem).toBe('0000007b5c9c61c332380f78');

    await orm.em.persistAndFlush(dataContainer);

    expect(dataContainer.data.partnerIdFromAnotherSystem).not.toBeInstanceOf(ObjectId);

    // need to clear to lost the original typing 'string'
    orm.em.clear();

    const getDataContainer = await orm.em.findOne(DataContainer, { id: dataContainer.id });
    expect(getDataContainer!.data.partnerIdFromAnotherSystem).not.toBeInstanceOf(ObjectId);
});
```

**Expected behavior**
I don't know what would be the best solution for this case.
But it is an implicit behavior that cannot be customized.

**Versions**

| Dependency | Version |
| - | - |
| node | 12.13.0 |
| typescript | 3.6.5 |
| mikro-orm | 3.4.1 |
| your-driver | mongodb |

**Comment by B4nan** [comment]
Hmm right, internally `MongoConnection` will convert everything that looks like an ObjectId to a real ObjectId, also for the insert/update payload. While this is fine/wanted for where conditions, we should convert only PK and FKs in the payload.

**Comment by iamomiid** [comment]
It still keeps happening for read payload! A simple `findOne` would reproduce the problem.

**Comment by B4nan** [comment]
There is a test case for this issue and its passing so please provide a new reproduction if its still happening.

**Comment by iamomiid** [comment]
Here you can check [this repository](https://github.com/iamomiid/mikro-orm-test-string). I wonder how the test case is passing. Note: Use `yarn start` to test.

**Comment by B4nan** [comment]
Ok I see, that is another issue. Your problem is with querying, where we still convert to object ids, the property is correctly stored as string in the db.

**Comment by B4nan** [comment]
will be fixed in v3.6.12
